src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs

```diff
@@ -15,82 +16,122 @@ namespace NuGet.Packaging.Rules
     internal class UpholdBuildConventionRule : IPackageRule
     {
         private static ManagedCodeConventions ManagedCodeConventions = new ManagedCodeConventions(new RuntimeGraph());
-        string[] _folders = new string[]
+        private static readonly IReadOnlyList<string> BuildFolders = new string[]
         {
             PackagingConstants.Folders.Build,
             PackagingConstants.Folders.BuildTransitive,
-            PackagingConstants.Folders.BuildCrossTargeting
+            PackagingConstants.Folders.BuildCrossTargeting,
+            "buildMultiTargeting"
         };
 
         public string MessageFormat => AnalysisResources.BuildConventionIsViolatedWarning;
 
         public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
         {
-            var warnings = new List<PackagingLogMessage>();
-            foreach (var folder in _folders)
+            var packageId = builder.NuspecReader.GetId();
+            var filesInPackage = builder.GetFiles();
+
+            var violations = IdentifyViolators(filesInPackage, packageId);
+            var warning = GenerateWarning(violations);
+            return warning == null
+                ? Array.Empty<PackagingLogMessage>()
+                : new[] { warning };
+        }
+
+        internal PackagingLogMessage GenerateWarning(ICollection<ConventionViolator> conventionViolators)
+        {
+            if (conventionViolators.Count == 0)
+            {
+                return null;
+            }
+
+            var warningMessage = new StringBuilder();
+            foreach (var conViolator in conventionViolators)
             {
-                var files = builder.GetFiles().Where(t => t.StartsWith(folder));
-                var packageId = builder.NuspecReader.GetId();
-                var conventionViolators = IdentifyViolators(files, packageId);
-                warnings.AddRange(GenerateWarnings(conventionViolators));
+                warningMessage.AppendLine(string.Format(MessageFormat, conViolator.Extension, conViolator.Path, conViolator.ExpectedPath));
             }
-            return warnings;
+
+            var message = PackagingLogMessage.CreateWarning(warningMessage.ToString(), NuGetLogCode.NU5129);
+            return message;
         }
 
-        internal IEnumerable<PackagingLogMessage> GenerateWarnings(IEnumerable<ConventionViolator> conventionViolators)
+        internal List<ConventionViolator> IdentifyViolators(IEnumerable<string> files, string packageId)
         {
+            var violations = new List<ConventionViolator>();
```

Comment:
I would prefer to use another variable name in all related places.

```suggestion
            var inconsistencies = new List<ConventionViolator>();
```